AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -359,35 +354,64 @@ - (void)deleteDocumentWithPartition:(NSString *)partition
       completionHandler([[MSDocumentWrapper alloc] initWithError:error documentId:documentId]);
       return;
     }
+    [self readFromLocalStorageWithPartition:partition
+                                 documentId:documentId
+                               documentType:[MSDictionaryDocument class]
+                          completionHandler:^(MSDocumentWrapper *_Nonnull document, MSTokensResponse *_Nonnull tokensResponse) {
+                            if ([self.reachability currentReachabilityStatus] == NotReachable || document.pendingOperation) {
+                              NSInteger deviceTimeToLive = writeOptions ? writeOptions.deviceTimeToLive : MSDataStoreTimeToLiveDefault;
```

Comment:
Any chance to move this block into separate procedure to make it more readable?

Reply:
We're still working on it, that's something we'd like to do (refactor code into smaller methods), thanks for the feedback :)